pandas/core/indexes/extension.py

```diff
@@ -223,29 +219,14 @@ def __getitem__(self, key):
         deprecate_ndim_indexing(result)
         return result
 
-    def __iter__(self):
-        return self._data.__iter__()
-
     # ---------------------------------------------------------------------
 
-    def __array__(self, dtype=None) -> np.ndarray:
-        return np.asarray(self._data, dtype=dtype)
-
     def _get_engine_target(self) -> np.ndarray:
         # NB: _values_for_argsort happens to match the desired engine targets
         #  for all of our existing EA-backed indexes, but in general
         #  cannot be relied upon to exist.
         return self._data._values_for_argsort()
 
-    @doc(Index.dropna)
-    def dropna(self, how="any"):
-        if how not in ("any", "all"):
-            raise ValueError(f"invalid how option: {how}")
-
-        if self.hasnans:
-            return self._shallow_copy(self._data[~self._isnan])
-        return self._shallow_copy()
```

Comment:
The only difference here with the Index one is the use of `self._data` vs `self._values`, which as far as I know shouldn't matter?

Reply:
not sure if it matters for this method, but the distinction would matter for MultiIndex, which does not have _data.

Reply:
In this case, MultiIndex actually overrides `dropna`, so that shouldn't even matter here. 

But indeed, in general it's only for MI that using `_values` vs `_data` matters, for all the others it's the same (I think?), which I suppose is the reason in the base class there is more usage of `_values`.